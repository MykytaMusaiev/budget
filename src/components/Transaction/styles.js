import styled from 'styled-components'

export const Wrapper = styled.div`
  background: ${({value}) => value <0 ? '#e2e1e1' : '#fff'};
  border: 1px solid #dbdbdb;
  border-radius: 3px;
  padding: 5px;
  margin: 5px 2px 5px 2px;
  display: flex;
`;
Wrapper.displayName = 'TransactionWrapper';

export const TransactionDate = styled.div`
    flex-grow: 1;
`
TransactionDate.displayName = 'TransactionDate'

export const Value = styled.div`
    flex-grow: 1;
`
TransactionDate.displayName = 'Value'

export const Comment = styled.div`
    flex-grow: 2;
`
TransactionDate.displayName = 'Comment'

