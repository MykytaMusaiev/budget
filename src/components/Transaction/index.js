import PropTypes from 'prop-types';
import {Comment, TransactionDate, Value, Wrapper} from "./styles";

const Transaction = ({transaction: {value, date, comment}}) => (
    <Wrapper value={value}>
        <TransactionDate>{date}</TransactionDate>
        <Value>Value: {Number(value).toFixed(2)}</Value>
        <Comment>Comment: {comment}</Comment>
        <br/>
    </Wrapper>
)

Transaction.propTypes = {
    transaction: PropTypes.shape({
        label: PropTypes.string,
        value: PropTypes.number
    })
}

Transaction.defaultProps = {
    transaction: {
        label: '',
        value: ''
    }
}

export default Transaction