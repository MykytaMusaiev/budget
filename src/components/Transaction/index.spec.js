import React from 'react'
import { shallow } from 'enzyme';
import Transaction from '.'

describe("Transaction component", () => {
    let sut;
    let props;

    beforeEach(() => {

        props = {
            transaction: {
                value: 23,
                comment: 'test',

                date: '01.01.2022'
            }
        }
        sut = shallow(<Transaction {...props}/>);
    })

    it('shout return true', ()=> {
        expect(true).toBe(true)
    })

    it('should show transaction', () => {
        expect(sut).toMatchSnapshot();
    })

    // it('should show 2 zeros after amount', () => {
    //     // const valueDiv = sut.find('Value').first()
    //     expect(sut.find('Value').text()).toBe('23.00')
    // })
})