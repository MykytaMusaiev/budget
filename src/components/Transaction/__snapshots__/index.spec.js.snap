// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should show transaction 1`] = `
<TransactionWrapper
  value={23}
>
  <Comment>
    01.01.2022
  </Comment>
  <styled.div>
    Value: 
    23.00
  </styled.div>
  <styled.div>
    Comment: 
    test
  </styled.div>
  <br />
</TransactionWrapper>
`;
